### ISSUE TITLE
Garry's Mod x64 hangs at launch on macOS Monterey 12.0.1

### ISSUE THREAD

**Issue by satanch** [Facepunch/garrysmod-issues#5111]
Garry's Mod hangs on blue loading screen or just jumping app icon on startup at macOS Monterey 12.0.1 (Apple M1).
Tried both branches (x64 and Chromium).
Attached crash dumps to this issue.

[crash_dumps_macos.zip](https://github.com/Facepunch/garrysmod-issues/files/7664126/crash_dumps_macos.zip)

**Comment by satanch** [comment]
Terminal output when I'm trying to launch it via binary

```
satanch@iMac-satanch MacOS % ./gmod
 failed to dlopen GarrysMod_Signed.app/Contents/MacOS//Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib error=dlopen(GarrysMod_Signed.app/Contents/MacOS//Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib, 0x0002): tried: 'GarrysMod_Signed.app/Contents/MacOS//Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib' (no such file), '/usr/lib/filesystem_stdio.dylib' (no such file)
 failed to dlopen GarrysMod.app/Contents/MacOS//Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib error=dlopen(GarrysMod.app/Contents/MacOS//Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib, 0x0002): tried: 'GarrysMod.app/Contents/MacOS//Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib' (no such file), '/usr/lib/filesystem_stdio.dylib' (no such file)
 failed to dlopen /Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib error=dlopen(/Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib, 0x0002): tried: '/Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib' (no such file), '/usr/lib/filesystem_stdio.dylib' (no such file)
Failed to load /Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib: (null)
AppFramework : Unable to load module /Users/satanch/Library/Application Support/Steam/steamapps/common/GarrysMod/filesystem_stdio.dylib!
** Unknown exception behavior: 0
```

**Comment by satanch** [comment]
**Little update**

This launch argument helped me to launch successfully:
```
-nochromium
````

**Comment by N0rdmann** [comment]
Just found this is already discussed here:
https://github.com/Facepunch/garrysmod-issues/issues/5001

Workaround to get a basic main menu with disabled chromium:
https://github.com/robotboy655/gmod-lua-menu

**Comment by satanch** [comment]
> Just found this is already discussed here: #5001
> 
> Workaround to get a basic main menu with disabled chromium: https://github.com/robotboy655/gmod-lua-menu

Unfortunately this addon doesn’t work :(

**Comment by N0rdmann** [comment]
Yeah didnt get the LUA Menu to work with x64 build neither, at least basic game is working, with console you can launch maps like: map gm_construct
Servers can be joined via steam serverbrowser and filter for game garrysmode...

**Comment by SageSwag** [comment]
I know its been about a month, but can you explain this in simpler terms? Im kinda confused 

> Yeah didnt get the LUA Menu to work with x64 build neither, at least basic game is working, with console you can launch maps like: map gm_construct Servers can be joined via steam serverbrowser and filter for game garrysmod...
> 
> **edit** The issue is with the finder default behaviour, if you choose migrate when copying **lua** + **materials** folders from gmod-lua-menu git to SteamApps/common/GarrysMod/garrysmod/ it does **NOT** replace garrysmod/lua/menu/menu.lua file (and therefore not loading the custom menu) - you can simply rename your original menu.lua to something else like menu.lua.old before copying over the 2 folders and it works...

**Comment by jordanshaikly** [comment]
Before extracting the LUA menu, rename the menu.lua that already exists to something else like menu.lua.old. 
Then it should work.

**Comment by EgorSerrator1** [comment]
> **Little update**

> This launch argument helped me to launch successfully:

> -nochromium


[satanch], So is it a solution? Have you understand how to solve the problem? Or is it unsolved?

**Comment by Jamescoll21** [comment]
> I know its been about a month, but can you explain this in simpler terms? Im kinda confused
> 
> > Yeah didnt get the LUA Menu to work with x64 build neither, at least basic game is working, with console you can launch maps like: map gm_construct Servers can be joined via steam serverbrowser and filter for game garrysmod...
> > **edit** The issue is with the finder default behaviour, if you choose migrate when copying **lua** + **materials** folders from gmod-lua-menu git to SteamApps/common/GarrysMod/garrysmod/ it does **NOT** replace garrysmod/lua/menu/menu.lua file (and therefore not loading the custom menu) - you can simply rename your original menu.lua to something else like menu.lua.old before copying over the 2 folders and it works...

https://youtu.be/8ak_g9JqSUU I’ve done a tutorial on how to apply it

**Comment by tkurgpold** [comment]
found a fix - some kind soul wrote a tool for updating CEF, and a couple weeks after the last post in this thread, they released a macOS version:
https://github.com/solsticegamestudios/GModCEFCodecFix
tested on a fresh install of garrys mod, without the lua menu - download the latest release, chmod +x to make it executable, takes about a minute to run, and after that the game gets to the main menu just fine.

**Comment by tenetms** [comment]
> found a fix - some kind soul wrote a tool for updating CEF, and a couple weeks after the last post in this thread, they released a macOS version: https://github.com/solsticegamestudios/GModCEFCodecFix tested on a fresh install of garrys mod, without the lua menu - download the latest release, chmod +x to make it executable, takes about a minute to run, and after that the game gets to the main menu just fine.

This method works flawlessly, the game works exactly as expected after applying the CEFCodecFix. Can the patched version be pushed into production?

**Comment by Steintree** [comment]
> > found a fix - some kind soul wrote a tool for updating CEF, and a couple weeks after the last post in this thread, they released a macOS version: https://github.com/solsticegamestudios/GModCEFCodecFix tested on a fresh install of garrys mod, without the lua menu - download the latest release, chmod +x to make it executable, takes about a minute to run, and after that the game gets to the main menu just fine.
> 
> This method works flawlessly, the game works exactly as expected after applying the CEFCodecFix. Can the patched version be pushed into production?

So do all features of the game work, because when using the lua menu some things do not work. It would be great if you could answer since I am not sure if it is worth it to buy the game.

**Comment by Kefta** [comment]
Yes CEF Codec Fix fixes everything.

**Comment by Steintree** [comment]
> Yes CEF Codec Fix fixes everything.

So that means that I can also use dupes and saves? Also how many FPS do you get?

**Comment by Kefta** [comment]
It's an HTML/web render engine fix, it has nothing to do with dupes, saves, nor FPS.

**Comment by Steintree** [comment]
> It's an HTML/web render engine fix, it has nothing to do with dupes, saves (other than viewing those two in the Steam workshop which launches a window), nor FPS.

No these question were dedicated to the game Garry’s Mod itself when running.

**Comment by Steintree** [comment]
Is this legal and is the game launched through Steam?

**Comment by tkurgpold** [comment]
> > > found a fix - some kind soul wrote a tool for updating CEF, and a couple weeks after the last post in this thread, they released a macOS version: https://github.com/solsticegamestudios/GModCEFCodecFix tested on a fresh install of garrys mod, without the lua menu - download the latest release, chmod +x to make it executable, takes about a minute to run, and after that the game gets to the main menu just fine.
> > 
> > 
> > This method works flawlessly, the game works exactly as expected after applying the CEFCodecFix. Can the patched version be pushed into production?
> 
> So do all features of the game work, because when using the lua menu some things do not work. It would be great if you could answer since I am not sure if it is worth it to buy the game.

I mean, given that it’s just updating a component of the game to be compatible with Monterey, yeah. :P

**Comment by tkurgpold** [comment]
> Is this legal and is the game launched through Steam?

It’s legal. CEF is open source, it’d be.. really weird if this was somehow illegal, tbh. We’re all already technically modding the game by going on the workshop, updating an open source dependency isn’t really a grey area.

**Comment by Kefta** [comment]
> > Is this legal and is the game launched through Steam?
> 
> It’s legal. CEF is open source, it’d be.. really weird if this was somehow illegal, tbh. We’re all already technically modding the game by going on the workshop, updating an open source dependency isn’t really a grey area.

But you said "I am not sure if it is worth it to buy the game" implying you may not have bought it. Also, modding the game is not illegal, there's no grey area, it's allowed.

**Comment by Steintree** [comment]
Thank you all for your helpful answers.
When reading the instruction to install the fix it looks like you can only launch the game through terminal, is there a way to launch it through steam?

**Comment by tkurgpold** [comment]
> Thank you all for your helpful answers.
> When reading the instruction to install the fix it looks like you can only launch the game through terminal, is there a way to launch it using Steam?

You don’t have to launch the game through terminal. I don’t even remember a way to launch the game through terminal being mentioned in the instructions.

**Comment by tkurgpold** [comment]
> > > Is this legal and is the game launched through Steam?
> > 
> > 
> > It’s legal. CEF is open source, it’d be.. really weird if this was somehow illegal, tbh. We’re all already technically modding the game by going on the workshop, updating an open source dependency isn’t really a grey area.
> 
> Modding the game is not illegal, there's no grey area, it's allowed.

I’m not sure why this was directed at me, but thanks for reinforcing my point

**Comment by Steintree** [comment]
> > Thank you all for your helpful answers.
> > When reading the instruction to install the fix it looks like you can only launch the game through terminal, is there a way to launch it using Steam?
> 
> You don’t have to launch the game through terminal. I don’t even remember a way to launch the game through terminal being mentioned in the instructions, tbh.

Well in step 9 it says that you should enter y in the terminal to launch Garry‘s Mod. https://www.solsticegamestudios.com/forums/threads/tutorial-gmod-cef-codec-fix-macos.1266/

**Comment by Steintree** [comment]
It says that it will not work if you run it through Steam.

**Comment by bigdogmat** [comment]
> Well in step 9 it says that you should enter y in the terminal to launch Garry‘s Mod.

It gives you the option to run Gmod when it finishes patching, not that you are required to run it through it.

It simple says some methods of launching will run a file validation thus reverting the changes made by the patch. 

> be sure to use the GModCEFCodecFix script to launch GMod, or launch GMod via your games list

Seems like launching it through the list of games in steam should work fine

**Comment by Kartoffeltoaster** [comment]
So right now I am installing the fix and it takes hours. I really don't know if this is normal.
Can anybody help me?

**Comment by Kartoffeltoaster** [comment]
> > Well in step 9 it says that you should enter y in the terminal to launch Garry‘s Mod.
> 
> It gives you the option to run Gmod when it finishes patching, not that you are required to run it through it.
> 
> It simple says some methods of launching will run a file validation thus reverting the changes made by the patch.
> 
> > be sure to use the GModCEFCodecFix script to launch GMod, or launch GMod via your games list
> 
> Seems like launching it through the list of games in steam should work fine


So right now I am installing the fix and it takes hours. I really don't know if this is normal.
Can anybody help me?

**Comment by Kartoffeltoaster** [comment]
I do not know how, but somehow it take hours to install.
Can anyone help me?

**Comment by Kartoffeltoaster** [comment]
> > Well in step 9 it says that you should enter y in the terminal to launch Garry‘s Mod.
> 
> It gives you the option to run Gmod when it finishes patching, not that you are required to run it through it.
> 
> It simple says some methods of launching will run a file validation thus reverting the changes made by the patch.
> 
> > be sure to use the GModCEFCodecFix script to launch GMod, or launch GMod via your games list
> 
> Seems like launching it through the list of games in steam should work fine

I do not know why, but somehow the fix takes hours to apply.
Can somebody help me?

**Comment by Kartoffeltoaster** [comment]
I do not know, but somehow I can not use any addons.

**Comment by Kartoffeltoaster** [comment]
> > found a fix - some kind soul wrote a tool for updating CEF, and a couple weeks after the last post in this thread, they released a macOS version: https://github.com/solsticegamestudios/GModCEFCodecFix tested on a fresh install of garrys mod, without the lua menu - download the latest release, chmod +x to make it executable, takes about a minute to run, and after that the game gets to the main menu just fine.
> 
> This method works flawlessly, the game works exactly as expected after applying the CEFCodecFix. Can the patched version be pushed into production?

It does not work flawlessly for me. It worked fine at first, then I closed the game and reopened it, but somehow it does not work.

**Comment by mattswain001** [comment]
Ok so I'm using GModCEF and in the launch options, I previously had -nochromium which made it load, but the menu GUI was missing so I had simply a background of pictures. Removing -nochromium from the launch options causes Good to crash immediately when launching. Have I done something wrong?

**Comment by hugsnugs** [comment]
can someone please make a video tutorial.

**Comment by jadenquinn** [comment]
This STILL does not work for me. Crashes immediately upon start, launching with -nochromium now does exactly what I was trying to fix before, not even getting to the title screen. This seems to have made things WORSE.
Has someone fixed this yet?

**Comment by mattswain001** [comment]
> This STILL does not work for me. Crashes immediately upon start, launching with -nochromium now does exactly what I was trying to fix before, not even getting to the title screen. This seems to have made things WORSE.
> Has someone fixed this yet? I've been trying all f*cking day.

Ok that's really weird. I can only suggest deleting gmod and all its files entirely, reinstalling and maybe trying again.

**Comment by jadenquinn** [comment]
> > This STILL does not work for me. Crashes immediately upon start, launching with -nochromium now does exactly what I was trying to fix before, not even getting to the title screen. This seems to have made things WORSE.
> > Has someone fixed this yet? I've been trying all f*cking day.
> 
> Ok that's really weird. I can only suggest deleting gmod and all its files entirely, reinstalling and maybe trying again.

I've done that 3 times, and once just copied the files from a completely other Mac.

**Comment by jadenquinn** [comment]
> > This STILL does not work for me. Crashes immediately upon start, launching with -nochromium now does exactly what I was trying to fix before, not even getting to the title screen. This seems to have made things WORSE.
> > Has someone fixed this yet? I've been trying all f*cking day.
> 
> Ok that's really weird. I can only suggest deleting gmod and all its files entirely, reinstalling and maybe trying again.

Alright, did it again, ran the fix tool thingy, and it doesn't even start to the title screen anymore.

**Comment by jadenquinn** [comment]
This is the log I get in terminal: https://pastebin.com/BRW5SQYN

**Comment by Purplous** [comment]
GUYS I FIXED IT AND I GOT A MENU THAT LETS YOU USE EVERYTHING
ITS A GMOD CUSTOM MENU AND IT WORKS SOOOOOO GOOD (as far as im aware) YOU DONT NEED -nochromium YOU JUST GOTTA INSTALL IT AND IT WORKS
HERES THE LINK:
https://www.reddit.com/r/gmod/comments/empq5c/release_gmod_custom_menu_gcm/

**Comment by Purplous** [comment]
Upon testing it appears to change everything in the spawn menu so Veichles is now spawnmenu.veichles for example im gonna just mess around a bit and attempt to Fix this

**Comment by Purplous** [comment]
UPDATE: just dont change the localisation files.

**Comment by iRYO400** [comment]
Sum up in 2022/09:
- Set launch mode to `x86-64 - Chromium + 64-bit binaries`
- Add this in launch options `-nochromium`
- Then do the things described here https://github.com/robotboy655/gmod-lua-menu ( if it doesn't, try to verify integrity, then try again)

Running on Monterey 12.5 / M1 Pro

**Comment by gomza22** [comment]
Hi guys, I have set the launch to x86-64 - Chromium + 64-bit binaries, and changed the original 'menu.lua' file to 'menu.lua.old', and after that merged the two necessary files to the garrysmod folder, and I am still am stuck with the blue screen with the black loading box at the bottom. Please please may someone suggest what else I could do to get in GMod.

**Comment by gomza22** [comment]
> localisation

what do you mean by this exactly? should i not tamper with any files and just add the downloaded files?

**Comment by Tarantula09** [comment]
ive tried every way to launch gmod suggested in this post with new installations of gmod for each method and sometimes combined and each method got me to the menu but there are no buttons

**Comment by Tarantula09** [comment]
> GUYS I FIXED IT AND I GOT A MENU THAT LETS YOU USE EVERYTHING ITS A GMOD CUSTOM MENU AND IT WORKS SOOOOOO GOOD (as far as im aware) YOU DONT NEED -nochromium YOU JUST GOTTA INSTALL IT AND IT WORKS HERES THE LINK: https://www.reddit.com/r/gmod/comments/empq5c/release_gmod_custom_menu_gcm/

well ive completely broke it now i cant figure out how to remove the gcm please reply!!!!

**Comment by Tarantula09** [comment]
nvm i got it working i installed every option that was discussed and it works flawlessly

**Comment by Tarantula09** [comment]
menu and eveything

**Comment by Blackflexxy** [comment]
I tried all of that, but it didn't work, clean install of gmod without any workshop items and nothing... https://youtu.be/xOkmXEi8q7M

**Comment by Kwuuk** [comment]
The codec fix is making my GMod crash immediately
